data-prepper-plugins/parse-json-processor/src/test/java/org/opensearch/dataprepper/plugins/codec/json/JsonInputCodecTest.java

```diff
@@ -215,6 +220,38 @@ void parse_with_InputStream_calls_Consumer_with_Event(final int numberOfObjects)
         }
     }
 
+    @ParameterizedTest
+    @ValueSource(ints = {1, 2, 10, 100})
+    void parse_with_InputStream_calls_Consumer_with_EventConfig(final int numberOfObjects) throws IOException {
```

Comment:
I think ideally, each of these conditions is tested independently of each other. I see three things: 1. Using a key_name, 2. Keys as data, 3. Keys as metadata.

Reply:
Thanks @dlvenable. I have added tests to cover the scenarios for 3 configs separately.